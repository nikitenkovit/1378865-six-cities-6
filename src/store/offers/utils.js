import {SortingType} from "../../const";

export const sortingFunction = (offers, type) => (a, b) => {
  switch (type) {
    case SortingType.HIGH_TO_LOW:
      return b.price - a.price;
    case SortingType.LOW_TO_HIGH:
      return a.price - b.price;
    case SortingType.TOP_RATED:
      return b.rating - a.rating;
  }
  return offers;
};
