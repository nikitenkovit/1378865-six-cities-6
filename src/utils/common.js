import offersData from '../mocks/offersData';
import {SortingType} from "../const";

export const getRatingStarValue = (number) => {
  return Math.round(number) * 20 + `%`;
};

export const adaptOfferData = (data) => ({
  bedrooms: data.bedrooms,
  city: data.city,
  description: data.description,
  goods: data.goods,
  host: {
    avatarUrl: data.host.avatar_url,
    id: data.host.id,
    isPro: data.host.is_pro,
    name: data.host.name,
  },
  id: data.id,
  images: data.images,
  isFavorite: data.is_favorite,
  isPremium: data.is_premium,
  location: data.location,
  maxAdults: data.max_adults,
  previewImage: data.preview_image,
  price: data.price,
  rating: data.rating,
  title: data.title,
  type: data.type,
});

export const adaptCommentsData = (data) => ({
  comment: data.comment,
  date: data.date,
  id: data.id,
  rating: data.rating,
  user: {
    avatarUrl: data.user.avatar_url,
    id: data.user.id,
    isPro: data.user.is_pro,
    name: data.user.name,
  }
});

export const getOffers = () => offersData.map(adaptOfferData); // временно. пока нет api

export const getOffersByCity = (state) => {
  return getOffers().filter((offer) => offer.city.name === state.CITY.name);
};

export const sortingFunction = (offers, type) => (a, b) => {
  switch (type) {
    case SortingType.HIGH_TO_LOW:
      return b.price - a.price;
    case SortingType.LOW_TO_HIGH:
      return a.price - b.price;
    case SortingType.TOP_RATED:
      return b.rating - a.rating;
  }
  return offers;
};
