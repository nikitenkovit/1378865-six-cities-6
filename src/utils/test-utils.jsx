import React from "react";
import {Router} from 'react-router-dom';
import {render as pureRender} from "@testing-library/react";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import browserHistory from "../history";

const render = (
    ui, {
      store = {},
      ...otherOptions
    }) => {
  const wrapper = ({children}) => (
    <Provider store={configureStore()(store)}>
      <Router history={browserHistory}>
        {children}
      </Router>
    </Provider>
  );
  return pureRender(ui, {wrapper, ...otherOptions});
};

// Реэкспортим библиотеку, чтобы не нужно было ее подключать.
export * from '@testing-library/react';
export {render};
