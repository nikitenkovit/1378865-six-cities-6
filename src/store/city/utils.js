import {DEFAULT_CURRENT_CITY} from "../../const";
import {getOffers} from "../../utils/common";

export const getCurrentCity = (state) => {
  if (state && state.CITY) {
    return state.CITY;
  }
  return getCities()
    .find((city) => city.name === DEFAULT_CURRENT_CITY);
};

export const getCities = () => {
  const cities = getOffers()
    .reduce((generalOffer, offer) => {
      if (!generalOffer.hasOwnProperty(offer.city.name)) {
        generalOffer[offer.city.name] = {...offer.city};
      }

      return generalOffer;
    }, {});

  return Object.values(cities);
};
