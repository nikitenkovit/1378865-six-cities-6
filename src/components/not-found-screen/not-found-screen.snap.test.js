import React from "react";
import {render} from "../../utils/test-utils";
import NotFoundScreen from "./not-found-screen";

const testStore = {
  USER: {
    user: {
      email: `test@test.ru`,
      name: `Name`,
    }
  }
};

describe(`NotFoundScreen tests`, () => {
  it(`Should NotFoundScreen render correctly`, () => {
    const {container} = render(
        <NotFoundScreen/>,
        {store: testStore},
    );
    expect(container).toMatchSnapshot();
  });
});
