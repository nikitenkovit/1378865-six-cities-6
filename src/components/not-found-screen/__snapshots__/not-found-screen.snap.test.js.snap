// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundScreen tests Should NotFoundScreen render correctly 1`] = `
<div>
  <div
    class="page page--favorites-empty"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
              style="flex-direction: column; align-items: flex-end;"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                    style="background-image: url(undefined);"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    test@test.ru
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--favorites page__main--favorites-empty"
    >
      <div
        class="page__favorites-container container"
      >
        <section
          class="favorites favorites--empty"
        >
          <h1
            class="visually-hidden"
          >
            Page not found
          </h1>
          <div
            class="favorites__status-wrapper"
          >
            <b
              class="favorites__status"
            >
              404. Page not found
            </b>
            <p
              class="favorites__status-description"
            >
              <a
                class="error-link"
                href="/"
                style="color: rgb(68, 129, 195); font-weight: bold;"
              >
                Go back to the home page
              </a>
            </p>
          </div>
        </section>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;
